### PR TITLE
Ref key case

### PR DESCRIPTION
**Pull request by atrites**
Adding an option to specify how the ref keys are cased. This will allow us to do case insensitive lookups.

### CODE REVIEW

File: lib/schema.utils.js [hunk 1]
```diff
@@ -99,7 +99,13 @@ exports.getIndexes = function (descriptor) {
           indexType = 'single';
         }
 
-        inds.push({ path, name, indexType });
+        let indData = { path, name, indexType };
+        let refKeyCase;
```
Comment (jcurtis) [inline lib/schema.utils.js]: Doesn't look used?

File: lib/schema.utils.js [hunk 2]
```diff
@@ -99,7 +99,13 @@ exports.getIndexes = function (descriptor) {
           indexType = 'single';
         }
 
-        inds.push({ path, name, indexType });
+        let indData = { path, name, indexType };
```
Comment (jcurtis) [inline lib/schema.utils.js]: `const`

Comment (atrites) [reply lib/schema.utils.js]: Conditional add to it just below

File: lib/schema.utils.js (hunk 1 above)
Comment (atrites) [reply lib/schema.utils.js]: ... true
Symptom of a refactor

File: lib/schema.utils.js (hunk 2 above)
Comment (jcurtis) [reply lib/schema.utils.js]: Still `const`. Only `let` when you reassign the variable entirely.

**Review by scdowney — APPROVED** [review]

**Review by bojand — APPROVED** [review]

**Pull request merged**